data/src/main/java/me/androidbox/data/local/agenda/AgendaLocalRepositoryImp.kt

```diff
@@ -0,0 +1,124 @@
+package me.androidbox.data.local.agenda
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+import me.androidbox.data.local.dao.EventDao
+import me.androidbox.data.local.dao.ReminderDao
+import me.androidbox.data.local.dao.TaskDao
+import me.androidbox.data.mapper.*
+import me.androidbox.domain.agenda.model.Agenda
+import me.androidbox.domain.agenda.model.Event
+import me.androidbox.domain.agenda.model.Reminder
+import me.androidbox.domain.agenda.model.Task
+import me.androidbox.domain.authentication.ResponseState
+import me.androidbox.domain.authentication.remote.AgendaLocalRepository
+import me.androidbox.domain.repository.AgendaRemoteRepository
+import java.time.ZoneId
+import javax.inject.Inject
+
+class AgendaLocalRepositoryImp @Inject constructor(
+    private val eventDao: EventDao,
+    private val taskDao: TaskDao,
+    private val reminderDao: ReminderDao,
+    private val agendaRemoteRepository: AgendaRemoteRepository
+) : AgendaLocalRepository {
+
+    override fun fetchAgenda(startTimeStamp: Long, endTimeStamp: Long): Flow<ResponseState<Agenda>> {
+        return flow<ResponseState<Agenda>> {
+
+            /* Fetch all the events, tasks, and reminders from the DB and return to populate the agenda screen */
+            val events = fetchEventsFromLocal(startTimeStamp, endTimeStamp)
+            val tasks = fetchTasksFromLocal(startTimeStamp, endTimeStamp)
+            val reminders = fetchRemindersFromLocal(startTimeStamp, endTimeStamp)
+
+            /* Emit after fetching from the DB */
+            emit(ResponseState.Success(Agenda(
+                events = events,
+                tasks = tasks,
+                reminders = reminders
+            )))
+
+            /* Fetch all agenda for the current day and time zone */
+            val agenda = agendaRemoteRepository.fetchAgendaForDay(
+                ZoneId.systemDefault(),
+                startTimeStamp * 1000
+            )
```

Comment:
@MKiperszmid I think there is something wrong. As I have events for the date I am fetching for. But the endpoint always returns empty. Is this correct?

Reply:
Do you have stuff on the api?
If so, is the timeStamp already in ms? if so, try removing the "* 1000"

Reply:
I use this method to get the start of the current day?   
```
private fun getStartOffCurrentDay(agendaDate: ZonedDateTime = ZonedDateTime.now(ZoneId.systemDefault())): ZonedDateTime {
        return agendaDate
            .toLocalDate()
            .atStartOfDay(ZoneId.systemDefault())
    }
```

Then I call it like this:
`agendaLocalRepository.fetchAgenda(getStartOffCurrentDay(agendaDate).toEpochSecond(), getEndOfCurrentDay(agendaDate).toEpochSecond())`

Which I use toEpochSecond() to change it to seconds

Then I change it to milli seconds by multipying by 1000 
```
            val agenda = agendaRemoteRepository.fetchAgendaForDay(
                ZoneId.systemDefault(),
                startTimeStamp * 1000
            )

I should have items on the BE for that day as they have already been upload every time I save to the db I use work manager to upload them. 
```

Reply:
Have you verified this, by checking the API with something like Postman?
Mostly to verify that you are indeed not getting any items from the API because an issue with your code, and not that the api is empty

Reply:
![Screenshot from 2023-04-14 02-18-18](https://user-images.githubusercontent.com/14260802/231861157-6ab41b27-0ae5-4648-aeae-7fb61d38b8c6.png)

Reply:
![Screenshot from 2023-04-14 02-18-18](https://user-images.githubusercontent.com/14260802/231861157-6ab41b27-0ae5-4648-aeae-7fb61d38b8c6.png)

Reply:
Then issue is that you aren't uploading the information correctly.
Have you tried uploading a Task or a Reminder just to verify it works?